apps/api/src/hyp3_api/validation.py

```diff
@@ -236,9 +232,33 @@ def check_opera_rtc_s1_static_coverage(job: dict, _) -> None:
 
     granule = granules[0]
     if not _has_opera_rtc_s1_static_coverage(granule):
-        raise GranuleValidationError(
-            f'Granule {granule} is outside the valid processing extent for OPERA RTC-S1 products.'
-        )
+        raise ValidationError(f'Granule {granule} is outside the valid processing extent for OPERA RTC-S1 products.')
+
+
+def check_aria_s1_gunw_dates(job: dict, _) -> None:
+    def get_date_from_job(date_param_key: str) -> date:
+        param_date_str = job['job_parameters'][date_param_key]
+        return date.fromisoformat(param_date_str)
+
+    job_dates = {
+        'reference_date': get_date_from_job('reference_date'),
+        'secondary_date': get_date_from_job('secondary_date'),
+    }
+
+    s1_start_date = date(2014, 6, 15)
+    todays_date = date.today()
+
+    for job_date_key, job_date in job_dates.items():
+        if job_date > todays_date:
+            raise ValidationError(f'"{job_date_key}" is {job_date} which is a date in the future.')
+
+        if job_date < s1_start_date:
+            raise ValidationError(
+                f'"{job_date_key}" is {job_date} which is before the start of the sentinel 1 mission ({s1_start_date}).'
+            )
```

Comment:
It might be more clear to factor out a helper function for handling the code that has to be run for both dates, e.g. `_check_aria_s1_gunw_date(job_params: dict, date_key: str)`, and then call that function for both date keys.